atom/browser/native_browser_view_mac.mm

```diff
@@ -51,6 +141,47 @@
   view.layer.backgroundColor = skia::CGColorCreateFromSkColor(color);
 }
 
+void NativeBrowserViewMac::UpdateDraggableRegions(
+    std::vector<gfx::Rect> system_drag_exclude_areas) {
+  NSView* webView = GetInspectableWebContentsView()->GetNativeView();
+  NSInteger webViewHeight = NSHeight([webView bounds]);
+  NSInteger webViewWidth = NSWidth([webView bounds]);
+
+  // Remove all DraggableRegionViews that are added last time.
+  // Note that [webView subviews] returns the view's mutable internal array and
+  // it should be copied to avoid mutating the original array while enumerating
+  // it.
+  base::scoped_nsobject<NSArray> subviews([[webView subviews] copy]);
+  for (NSView* subview in subviews.get())
+    if ([subview isKindOfClass:[DragRegionView class]])
+      [subview removeFromSuperview];
+
+  // Create one giant NSView that is draggable.
+  base::scoped_nsobject<NSView> dragRegion(
+        [[DragRegionView alloc] initWithFrame:NSZeroRect]);
+    [dragRegion setFrame:NSMakeRect(0,
+                                    0,
+                                    webViewWidth,
+                                    webViewHeight)];
+
+  // Then, on top of that, add "exclusion zones"
+  for (std::vector<gfx::Rect>::const_iterator iter =
+           system_drag_exclude_areas.begin();
+       iter != system_drag_exclude_areas.end();
+       ++iter) {
+    base::scoped_nsobject<NSView> controlRegion(
+        [[ExcludeDragRegionView alloc] initWithFrame:NSZeroRect]);
+    [controlRegion setFrame:NSMakeRect(iter->x(),
```

Comment:
There is a good chance I'm missing something, but just in case:  is this assuming the browser view bounds are 100% of the containing window?  Should use `view.frame` and `view.superview.frame` to handle the case where the `browserview.SetBounds()` has been used to position it in the bottom right corner?